jbpm-human-task/jbpm-human-task-jpa/src/main/java/org/jbpm/services/task/persistence/JPATaskPersistenceContext.java

```diff
@@ -225,12 +227,49 @@ public User updateUser(User user) {
 	}
 
 	@Override
-	public User removeUser(User user) {
+    public User removeUser(User user) {
 		check();
-		em.remove( user );
-		return user;
+        em.remove(user);
+        return user;
 	}
 
+    @Override
+    public Email findEmail(String emailId) {
+        check();
+        if (this.pessimisticLocking) {
+            return this.em.find(EmailImpl.class, emailId, LockModeType.PESSIMISTIC_WRITE);
+        }
+        return this.em.find(EmailImpl.class, emailId);
+    }
+
+    @Override
+    public Email persistEmail(Email email) {
+        check();
+        try {
+            this.em.persist(email);
+            if (this.pessimisticLocking) {
+                this.em.flush();
+                return this.em.find(EmailImpl.class, email.getId(), LockModeType.PESSIMISTIC_WRITE);
+            }
+        } catch (EntityExistsException e) {
+            throw new RuntimeException("Email already exists with " + email + " id, please check that there is no group and user with same id");
+        }
+        return email;
+    }
+
+    @Override
+    public Email updateEmail(Email email) {
+        check();
+        return this.em.merge(email);
+    }
+
+    @Override
+    public Email removeEmail(Email email) {
+        check();
+        em.remove(email);
+        return email;
+    }
```

Comment:
I cannot see where these methods are invoked and if they are really needed.
For persist, at the end, _persistIfNotExist_ calls to _tpc.persistOrgEntity(entity);_
Could you clarify if they are needed for persistence? 
I cannot see an analogous scenario to _add-group-command_ for email.

Reply:
for consistency leting this here.